bundles/org.openhab.binding.myq/src/main/java/org/openhab/binding/myq/internal/handler/MyQAccountHandler.java

```diff
@@ -107,6 +109,24 @@ public void dispose() {
         stopPolls();
     }
 
+    @Override
+    public Collection<Class<? extends ThingHandlerService>> getServices() {
+        return Collections.singleton(MyQDiscoveryService.class);
+    }
+
+    @Override
+    public void childHandlerInitialized(ThingHandler childHandler, Thing childThing) {
+        DevicesDTO localDeviceCaches = devicesCache;
+        if (localDeviceCaches != null) {
+            DeviceDTO device = localDeviceCaches.items.stream()
+                    .filter(d -> childHandler.getThing().getUID().getId().equalsIgnoreCase(d.serialNumber)).findFirst()
```

Comment:
You're using the id as a unique identifier.  However the convention is that the user can give it any name allowed. So it should not be used as identifier. You should check here on the property `serialNumber`.

Reply:
Good catch !